src/renderer/actions/utils/kill.js

```diff
@@ -26,12 +27,67 @@ export default async function killProcesses(channels, progressPerLoop, shouldRes
             const discordPid = results.find(p => parentPids.includes(p.pid));
             const bin = process.platform === "darwin" ? path.resolve(discordPid.bin, "..", "..", "..") : discordPid.bin;
             await new Promise(r => kill(discordPid.pid, r));
-            if (shouldRestart) setTimeout(() => shell.openPath(bin), 1000);
+            if (shouldRestart) {
```

Comment:
This whole section feels really hacky and susceptible to breaking.

Reply:
It's the only way to identify the applications. And don't worry, it's not hacky whatsoever. Linux has completely standardized paths that must be adhered to:

- `/bin` and `/usr/bin` and `/usr/local/bin` - This is where native packages install themselves.
- `/app` - This is Flatpak's virtual filesystem prefix, and is hardcoded inside the Flatpak standard.
- `/snap` - This is Snap's virtual filesystem prefix, and is hardcoded inside Snap.

There is no other way on Linux to identify if an app was launched as a Flatpak, Snap or Native. I must look at the binary paths. Because their sandboxed processes run without any "flatpak" or "snap" parent process identifiers.

But yeah, don't worry. It's all standardized and won't break. Also remember the inertia of big projects: They don't change trivial things like their `/app` and `/snap` prefixes just for a laugh. They have ZERO reason to ever change their prefix. If they do, we'll just update BD-Installer of course. But the chance of that ever happening is about 0.0%. :)